device/noise-protocol.go

```diff
@@ -16,14 +17,33 @@ import (
 	"golang.zx2c4.com/wireguard/tai64n"
 )
 
+type handshakeState int
+
 const (
-	HandshakeZeroed = iota
-	HandshakeInitiationCreated
-	HandshakeInitiationConsumed
-	HandshakeResponseCreated
-	HandshakeResponseConsumed
+	handshakeZeroed = handshakeState(iota)
```

Comment:
Can you add a TODO to this const block to explain each state and what the legal transitions are? It's obvious to anyone very familiar with WireGuard, but having the information inlined helps new people onboard.